lib/Doctrine/DBAL/Event/SchemaAlterTableAddColumnEventArgs.php

```diff
@@ -57,17 +58,15 @@ public function getPlatform()
     }
 
     /**
+     * Passing multiple SQL statements as an array is deprecated. Pass each statement as an individual argument instead.
+     *
      * @param string|string[] $sql
      *
      * @return \Doctrine\DBAL\Event\SchemaAlterTableAddColumnEventArgs
      */
     public function addSql($sql)
     {
-        if (is_array($sql)) {
-            $this->sql = array_merge($this->sql, $sql);
-        } else {
-            $this->sql[] = $sql;
-        }
+        $this->sql = array_merge($this->sql, is_array($sql) ? $sql : func_get_args());
```

Comment:
Maybe `(array) $sql`?

Reply:
Not sure I understand. The logic is that if the passed argument is an array (old behavior), we merge it into the property. Otherwise, we take all function arguments (future compatibility) and merge them.

What do you suggest to cast to an array? Could you suggest a code change?

Reply:
`$sql` is a `string|string[]`: an `(array)` cast will convert it to a `string[]`

Reply:
I can't see either how you could use a cast and still be able to detect whether you should handle the new variadicity (variadicism?) of this method. Please do suggest a code change.

Reply:
Oh, I see, the signature isn't `string|string[]`, but `string|string[]|...string`: I totally missed that!